odoo-openmrs/src/main/java/com/ozonehis/eip/odoo/openmrs/routes/MedicationRequestRouting.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright © 2024, Ozone HIS <info@ozone-his.com>
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.ozonehis.eip.odoo.openmrs.routes;
+
+import com.ozonehis.eip.odoo.openmrs.Constants;
+import com.ozonehis.eip.odoo.openmrs.processors.MedicationRequestProcessor;
+import lombok.Setter;
+import org.apache.camel.LoggingLevel;
+import org.apache.camel.builder.RouteBuilder;
+import org.hl7.fhir.r4.model.MedicationRequest;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Setter
+@Component
+public class MedicationRequestRouting extends RouteBuilder {
+
+    private static final String MEDICATION_REQUEST_ID = "medication.request.id";
+
+    private static final String MEDICATION_REQUEST_INCLUDE_PARAMS =
+            "MedicationRequest:encounter,MedicationRequest:medication,MedicationRequest:patient";
+
+    @Autowired
+    private MedicationRequestProcessor medicationRequestProcessor;
+
+    @Override
+    public void configure() {
+        // spotless:off
+        from("direct:fhir-medicationrequest")
+                .routeId("medication-request-to-sale-order-router")
+                .filter(body().isNotNull())
+                .filter(exchange -> exchange.getMessage().getBody() instanceof MedicationRequest)
+                .process(exchange -> {
+                    MedicationRequest medicationRequest = exchange.getMessage().getBody(MedicationRequest.class);
+                    exchange.setProperty(Constants.FHIR_RESOURCE_TYPE, medicationRequest.fhirType());
+                    exchange.setProperty(
+                            MEDICATION_REQUEST_ID,
+                            medicationRequest.getIdElement().getIdPart());
+                    exchange.getMessage().setBody(medicationRequest);
+                })
+                .toD("openmrs-fhir://?id=${exchangeProperty." + MEDICATION_REQUEST_ID + "}&resource=${exchangeProperty."
+                        + Constants.FHIR_RESOURCE_TYPE + "}&include=" + MEDICATION_REQUEST_INCLUDE_PARAMS)
```

Comment:
We should move this part to `camel-openmrs-fhir` project as it's the same logic for both ERPNext and Odoo, So there will an additional route that spits out medication request bundle.

Reply:
Agreed. Maybe we can take this up after this PR is merged.

Reply:
cc: @rbuisson